File: MATEMÁTICA/1161/main.js
function factorial(n, lookup) {
    if (n < BigInt(2)) {
        return BigInt(1);
    }
    if (! lookup[n]) {
        lookup[n] = n * factorial(n - BigInt(1), lookup);
    }
    return lookup[n];
}

let input = require('fs').readFileSync('/dev/stdin', 'utf-8');
let lines = input.split('\n');
let lookup = {}
for (let line of lines) {
    if (line) {
        let [m, n] = line.split(' ').map(BigInt);
        let result = factorial(m, lookup) + factorial(n, lookup);
        console.log(result.toString());
    }
}

File: MATEMÁTICA/1161/main.dart
import 'dart:io';

int factorial(int n, Map<int, int> lookup) {
  if (n < 2) return 1;
  if (lookup[n] == null) lookup[n] = n * factorial(n - 1, lookup);
  return lookup[n]!;
}

void main(List<String> args) {
  List<String>? line;
  Map<int, int> lookup = {};
  while ((line = stdin.readLineSync()?.split(' ')) != null) {
    int m = int.parse(line![0]);
    int n = int.parse(line[1]);
    print(factorial(m, lookup) + factorial(n, lookup));
  }
}
